faker/providers/internet/__init__.py

```diff
@@ -99,15 +101,23 @@ def url(self):
         pattern = self.random_element(self.url_formats)
         return self.generator.parse(pattern)
 
-    def ipv4(self):
-        """
-        Convert 32-bit integer to dotted IPv4 address.
-        """
-        return ".".join(map(lambda n: str(random.randint(-2147483648, 2147483647) >> n & 0xFF), [24, 16, 8, 0]))
-
-    def ipv6(self):
-        res = [hex(random.randint(0, 65535))[2:].zfill(4) for i in range(0, 8)]
-        return ":".join(res)
+    def ipv4(self, network=False):
+        """ Produce a random IPv4 address or network with a valid CIDR. """
+        address = str(ip_address(random.randint(
+            0, (2 ** IPV4LENGTH) - 1)))
+        if network:
+            address += '/' + str(random.randint(0, IPV4LENGTH))
+            address = str(ip_network(unicode(address), strict=False))
```

Comment:
See for yourself how `ipaddress` expect explicit unicode string:
```python
Python 2.7.11 (default, Dec  5 2015, 14:44:53) 
[GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.1.76)] on darwin
>>> from ipaddress import ip_network
>>> ip_network(unicode('49.50.51.52/22'), strict=False)
IPv4Network(u'49.50.48.0/22')
>>> ip_network('49.50.51.52/22', strict=False)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/kdeldycke/virtualenvs/billing/lib/python2.7/site-packages/ipaddress-1.0.16-py2.7.egg/ipaddress.py", line 199, in ip_network
ipaddress.AddressValueError: '49.50.51.52/22' does not appear to be an IPv4 or IPv6 network. Did you pass in a bytes (str in Python 2) instead of a unicode object?
>>> 
```